src/sinol_make/helpers/compile.py

```diff
@@ -71,15 +94,14 @@ def compile_file(file_path: str, name: str, compilers: Compilers, weak_compilati
 
     output = os.path.join(executable_dir, name)
     compile_log_path = os.path.join(compile_log_dir, os.path.splitext(name)[0] + '.compile_log')
-    compile_log = open(compile_log_path, 'w')
-
-    try:
-        if compile(file_path, output, compilers, compile_log, weak_compilation_flags):
-            return output, compile_log_path
-        else:
+    with open(compile_log_path, 'w') as compile_log:
+        try:
+            if compile(file_path, output, compilers, compile_log, weak_compilation_flags):
+                return output, compile_log_path
+            else:
+                return None, compile_log_path
+        except CompilationError:
             return None, compile_log_path
```

Comment:
```suggestion
        except CompilationError:
            pass
        return None, compile_log_path
```
Maybe it can be further simplified, I'm not familiar with exceptions in Python.